encoding/src/commonMain/kotlin/diglol/encoding/internal/commonBase45.kt

```diff
@@ -0,0 +1,135 @@
+package diglol.encoding.internal
+
+import kotlin.native.concurrent.SharedImmutable
+
+/**
+ * Implementation of Base45 encoding/decoding.
+ *
+ * Basically, Base45 is just encode 2 bytes based on 256 into 3 bytes based on 45.
+ *
+ * It will result in a encoded string which will generate a compat QR code when using alphanumeric mode.
+ *
+ * @see [RFC 9285](https://datatracker.ietf.org/doc/rfc9285/)
+ */
+
+@SharedImmutable
+internal val BASE45 = byteArrayOf(
+  48, 49, 50, 51, 52, 53, 54, 55, 56, 57, // 0-9
+  65, 66, 67, 68, 69, 70, 71, 72, 73, 74, // A-J
+  75, 76, 77, 78, 79, 80, 81, 82, 83, 84, // K-T
+  85, 86, 87, 88, 89, 90, // U-Z
+  // SP, $, %, *, +, -, ., /, :
+  32, 36, 37, 42, 43, 45, 46, 47, 58
+)
+
+
+internal fun ByteArray.commonEncodeBase45(map: ByteArray = BASE45): ByteArray {
+  val outSize = sizeOfEncodedBase45()
+  val outBytes = ByteArray(outSize)
+  val limit = if (size % 2 == 0) size else size - 1
+
+  var i = 0
+  var j = 0
+
+  while (i < limit) {
+    val a = this[i].toInt() and 0xff
+    val b = this[i + 1].toInt() and 0xff
+
+    val n = (a shl 8) + b
+
+    val c = n % 45
+    val d = (n / 45) % 45
+    val e = ((n / 45) / 45) % 45
+
+    outBytes[j] = map[c]
+    outBytes[j + 1] = map[d]
+    outBytes[j + 2] = map[e]
+
+    i += 2
+    j += 3
+  }
+
+  if (limit < size) {
+    val a = this[limit].toInt() and 0xff
+    val c = a % 45
+    val d = (a / 45) % 45
+    outBytes[j] = map[c]
+    outBytes[j + 1] = map[d]
+  }
+
+  return outBytes
+}
+
+private fun ByteArray.sizeOfEncodedBase45(): Int {
```

Comment:
Fixed in cda3eecd

Reply:
The same as `isEncodedSizeNotValid`.

Reply:
This method is a bit complicated.

If I manually inline this.

It will become 

```kotlin 
 val outSize = if (isEmpty()) {
    0
  } else {
    if (size % 2 == 0) {
      size * 3 / 2
    } else {
      (size - 1) * 3 / 2 + 2
    }
  }
```

reducing the readability.

What do you think?

Reply:
It‘s only used once, perhaps using `when` keyword would improve readability. 

```
val outSize = when {
  isEmpty() -> 0
  (size % 2 == 0) -> size * 3 / 2
  else -> (size - 1) * 3 / 2 + 2
}
```

Reply:
> It‘s only used once, perhaps using `when` keyword would improve readability.
> 
> ```
> val outSize = when {
>   isEmpty() -> 0
>   (size % 2 == 0) -> size * 3 / 2
>   else -> (size - 1) * 3 / 2 + 2
> }
> ```

ok, I will refactor to this approach.

Reply:
> It‘s only used once, perhaps using `when` keyword would improve readability.
> 
> ```
> val outSize = when {
>   isEmpty() -> 0
>   (size % 2 == 0) -> size * 3 / 2
>   else -> (size - 1) * 3 / 2 + 2
> }
> ```

Fixed in fc092998